build.gradle

```diff
@@ -82,39 +47,25 @@ sourceSets {
     }
     test {
         java {
-            srcDirs = ["src/test"]
+            srcDirs = ["src/test", "src/CommunityCommons/javasource"]
         }
     }
 }
 
-task copyToUserlib( type: Copy ) {
+task copyAllToUserlib( type: Copy ) {
+    from configurations.testCompileClasspath
+    exclude "public-api-${mendixVersion}.jar", "json-${mendixVersion}.jar", "logging-api-${mendixVersion}.jar", "m2ee-api-${mendixVersion}.jar"
     into userLibDir
-    from configurations.runtime
-    eachFile { fileCopyDetails ->
-        def requiredLibFlag = new File( destinationDir, "${fileCopyDetails.name}.${project.name}.RequiredLib")
-        requiredLibFlag.write ''
-    }
+    eachFile { fileCopyDetails -> new File( destinationDir, "${fileCopyDetails.name}.${project.name}.RequiredLib").write '' }
 }
 
-task untarMxbuild( type: Copy ) {
-    configurations.tar.findAll{it.name.endsWith('tar.gz')}.each {
-        from tarTree(resources.gzip(it))
-        into buildDir
-        include('**/runtime/bundles/com.mendix.public-api.jar')
-        include('**/runtime/bundles/com.mendix.logging-api.jar')
-        include('**/runtime/bundles/com.mendix.m2ee-api.jar')
-        include('**/runtime/bundles/com.mendix.json.jar')
-        include('**/runtime/bundles/javax.servlet-api.servlet.jar')
-        includeEmptyDirs = false
-    }
-}
-
-task prepareDeps {
-    dependsOn 'clean', 'copyToUserlib', 'untarMxbuild'
+task copyRuntimeToUserlib( type: Copy ) {

```

Comment:
The public-api JARs should be a compile-time only dependency. They're not part of the release package and as such shouldn't be copied to userlib.